libraries/chain/database_worker_objects.cpp

```diff
@@ -155,35 +155,79 @@ namespace golos { namespace chain {
 
         const auto& wto_idx = get_index<worker_techspec_index, by_next_cashout_time>();
 
-        for (auto wto_itr = wto_idx.begin(); wto_itr != wto_idx.end() && wto_itr->next_cashout_time <= now; ++wto_itr) {
-            const auto& wpo = get_worker_proposal(wto_itr->worker_proposal_author, wto_itr->worker_proposal_permlink);
+        // Updating consumption
+
+        asset consumption;
+
+        auto month_sec = fc::days(30).to_seconds();
+        for (auto wto_itr = wto_idx.begin(); wto_itr != wto_idx.end() && wto_itr->next_cashout_time < time_point_sec::maximum(); ++wto_itr) {
+            auto payment = (wto_itr->specification_cost + wto_itr->development_cost) / wto_itr->payments_count;
+            auto remaining = wto_itr->development_cost + wto_itr->specification_cost - (payment * wto_itr->finished_payments_count);
+            payment = remaining / (wto_itr->payments_count - wto_itr->finished_payments_count);
 
-            const auto& reward = wto_itr->development_cost / wto_itr->payments_count;
+            auto month_payments = std::min(month_sec / wto_itr->payments_interval, int64_t(wto_itr->payments_count - wto_itr->finished_payments_count));
```

Comment:
Very complex calculation. It can be more simple:
Total payment period: `payments_interval * payments_count`
Payment per second: `(development_cost + specification_cost) / total_payment_period`
Payment per month: `(development_cost + specification_cost) * fc::days(30).to_seconds() / (payments_interval * payments_count)`